solr/core/src/java/org/apache/solr/cli/SolrCLI.java

```diff
@@ -201,9 +201,10 @@ public static CommandLine parseCmdLine(String toolName, String[] args, List<Opti
   }
 
   public static String getDefaultSolrUrl() {
-    String scheme = EnvUtils.getEnv("SOLR_URL_SCHEME", "http");
-    String host = EnvUtils.getEnv("SOLR_TOOL_HOST", "localhost");
-    String port = EnvUtils.getEnv("SOLR_PORT", "8983");
+    // note that ENV_VAR syntax (and the env vars too) are mapped to env.var sys props
+    String scheme = EnvUtils.getProperty("SOLR_URL_SCHEME", "http");
+    String host = EnvUtils.getProperty("SOLR_TOOL_HOST", "localhost");
+    String port = EnvUtils.getProperty("SOLR_PORT", "8983");
```

Comment:
`getProperty` calls `getPropertyWithCamelCaseFallback` calls `camelCaseToDotSeparated` which converts, so it works.  But I'll change it.

Reply:
Hmm, didn't recall I was that clever :)

Reply:
I was mistaken; "camelCaseToDotSeparated" is not the same as "[screaming snake case](https://www.theserverside.com/definition/Snake-case)".  It could have used a commented example.